spark/v3.2/spark/src/test/java/org/apache/iceberg/spark/source/StructRecord.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.spark.source;
+
+import org.apache.iceberg.relocated.com.google.common.base.Objects;
+
+public class StructRecord {
```

Comment:
Nit: As these two classes (StructRecord and InnerRecord) seem to only be used as java beans during tests, could you possibly name them something a bit more specific / something that makes it more clear that these are for test purposes?

The file `StructRecord` on its own really seems like it is part of the framework and could easily be confusing for people (e.g. it was confusing for me).

Some of the Spark tests use `LogRecord`, as well as some use `SimpleRecord`. Admittedly `SimpleRecord` might have the same naming implications, but `StructRecord` is so close to existing concepts and classes like `StructReader` for me.

Maybe `TwoLevelRecord` or something might be more appropriate? I can see how `StructRecord` seems similar to `SimpleRecord` (though I personally wouldn't have used `SimpleRecord` either), but `StructRecord` is _super_ similar in name to existing Iceberg concepts.

Even just `ComplexRecord` or `NestedRecord` or even `TestStructRecord` would be a lot better in my opinion.

Reply:
Updated